google/oauth2/webauthn_handler.py

```diff
@@ -0,0 +1,218 @@
+        if self.timeout_ms:
+            req_options['timeout'] = self.timeout_ms
+        if self.allow_credentials:
+            req_options['allowCredentials'] = [c.to_dict() for c in self.allow_credentials]
+        if self.user_verification:
+            req_options['userVerification'] = self.user_verification
+        if self.extensions:
+            req_options['extensions'] = self.extensions.to_dict()
+        return json.dumps({
+            'type': 'get',
+            'origin': self.origin,
+            'requestData': req_options,
+        })
+
+@dataclass(frozen=True)
+class AuthenticatorAssertionResponse:
+    """Authenticator response to a WebAuthn get (assertion) request.
+
+    https://www.w3.org/TR/webauthn-3/#authenticatorassertionresponse
+    
+    Args:
+        client_data_json: <url-safe base64-encoded> client data JSON.
+        authenticator_data: <url-safe base64-encoded> authenticator data.
+        signature: <url-safe base64-encoded> signature.
+        user_handle: <url-safe base64-encoded> user handle.
+    """
+    client_data_json: str
+    authenticator_data: str
+    signature: str
+    user_handle: Optional[str]
+
+@dataclass(frozen=True)
+class GetResponse:
+    """WebAuthn get (assertion) response.
+    
+    Args:
+        id: <url-safe base64-encoded> credential id (key handle).
+        response: The authenticator assertion response.
+        authenticator_attachment: <'cross-platform'|'platform'> The attachment status of the authenticator.
+        client_extension_results: WebAuthn authentication extensions output results in a dictionary.
+    """
+    id: str
+    response: AuthenticatorAssertionResponse
+    authenticator_attachment: Optional[str]
+    client_extension_results: Optional[Dict]
+
+    @staticmethod
+    def from_json(json_str: str):
+        """Verify and construct GetResponse from a JSON string."""
+        try:
+            resp_json = json.loads(json_str)
+        except ValueError:
+            raise exceptions.MalformedError('Invalid Get JSON response')
+        if resp_json['type'] != 'getResponse':
+            raise exceptions.InvalidOperation('Invalid Get response type: {}'.format(resp_json['type']))
+        pk_cred = resp_json['responseData']
+        if pk_cred is None:
+            if resp_json['error']:
+                raise exceptions.ReauthFailError('WebAuthn.get failure: {}'.format(resp_json['error']))
+            else:
+                raise exceptions.InvalidValue('Get response is empty')
+        if pk_cred['type'] != 'public-key':
+            raise exceptions.InvalidValue('Invalid credential type: {}'.format(pk_cred['type']))
+        assertion_json = pk_cred['response']
+        assertion_resp = AuthenticatorAssertionResponse(
+            client_data_json=assertion_json['clientDataJSON'],
+            authenticator_data=assertion_json['authenticatorData'],
+            signature=assertion_json['signature'],
+            user_handle=assertion_json['userHandle'])
```

Comment:
Change to `user_handle=assertion_json.get('userHandle')` to return None if this value is not present in the response from the key.